idea/idea-fir/tests/org/jetbrains/kotlin/idea/quickfix/AbstractHighLevelQuickFixTest.kt

```diff
@@ -7,20 +7,90 @@ package org.jetbrains.kotlin.idea.quickfix
 
 import com.intellij.codeInsight.intention.IntentionAction
 import com.intellij.codeInspection.InspectionProfileEntry
+import com.intellij.openapi.util.io.FileUtil
 import org.jetbrains.kotlin.test.uitls.IgnoreTests
+import java.io.File
 import java.nio.file.Paths
 
 
 abstract class AbstractHighLevelQuickFixTest : AbstractQuickFixTest() {
+    companion object {
+        private val isTeamCityBuild: Boolean = System.getProperty("TEAMCITY_VERSION") != null
+    }
+
     override fun doTest(beforeFileName: String) {
         IgnoreTests.runTestIfEnabledByFileDirective(
             Paths.get(beforeFileName),
-            enableTestDirective = IgnoreTests.DIRECTIVES.FIR_COMPARISON_MUTLTILINE_COMMENT,
+            enableTestDirective = IgnoreTests.DIRECTIVES.FIR_COMPARISON_MULTILINE_COMMENT,
             directivePosition = IgnoreTests.DirectivePosition.LAST_LINE_IN_FILE,
             additionalFilesExtensions = arrayOf("after")
         ) {
             super.doTest(beforeFileName)
+            // getFir/OriginalAfterFile expect just the filename (no directory), but beforeFileName has the directory
+            val beforeFileNameOnly = File(beforeFileName).name
+            val firAfterFile = getFirAfterFile(beforeFileNameOnly)
+            val originalAfterFile = getOriginalAfterFile(beforeFileNameOnly)
+            if (firAfterFile.exists() && firAfterFile.readText().trim() == originalAfterFile.readText().trim()) {
+                // The FIR after file is identical with the original after file. Remove the FIR after file and mark "FIR_IDENTICAL" in the original files.
+                // Note: FIR_IDENTICAL cannot be at the start of the file because the test uses ActionHint which parses the first line to get
+                // the action to perform (or check that they are not available).
+                firAfterFile.delete()
+
+                fun appendFirIdentical(file: File) {
+                    // The file ends with FIR_COMPARISON directive. It has to remain the last line for the test to be enabled, FIR_IDENTICAL must go before that.
+                    val lines = file.readLines().dropLast(1)
+                    file.writer().use {
+                        for (line in lines) {
+                            it.appendLine(line)
+                        }
+                        it.appendLine(IgnoreTests.DIRECTIVES.FIR_IDENTICAL)
+                        it.appendLine(IgnoreTests.DIRECTIVES.FIR_COMPARISON_MULTILINE_COMMENT)
+                    }
+                }
+
+                val beforeFile = File(beforeFileName)
+                appendFirIdentical(beforeFile)
+                appendFirIdentical(originalAfterFile)
```

Comment:
If identical.... why do we need to add the directive to both files? Isn't it enough to have that in (original) after file, assuming that file records FE 1.0 behavior? What does `// FIR_IDENTICAL` mean in (original) before file?

Reply:
AFAICT, the test fixture takes the before file in an editor, runs any quick fixes, then compares the resulting file contents with the after file. So the directives have to be in all files (before, FE1.0 after, FIR after) so that the comparisons pass. After all, the directives are just comments which is why they should be there in all files.

Reply:
Ohhhhh, now I got it. Unlike that directive's usage in IrText tests and FirOldDiagnostics tests, where that directive just indicates either psi2ir v.s. fir2ir behavior (IrText output) match or FE 1.0 v.s. FIR diagnostics output match, that directive here is literally compared against file contents before/after quick fixes?!

Alright, for someone like me, please add a comment why that directive needs to be added to both. :)

Reply:
Yes, that is correct, it is literally compared. I would add a comment, but I took out the whole FIR_IDENTICAL bit anyway as per Ilya's suggestion 🙂